### PR TITLE
Compute and export total PM2.5 and PM10 for UFS

### PR DESCRIPTION
**Pull request by rmontuoro**
This PR introduces changes to the UFS/NUOPC cap to compute and export total diagnostic PM2.5 and PM10 for the UFS weather model. These quantities are computed according to formulas provided by @amdasilva.

The changes have been tested within UFS-GOCART on NOAA's Hera platform.

### CODE REVIEW

**Comment by mathomp4** [comment]
Whoops! Looks like I need to make some surgery on the CI. Give me a moment or two...

**Comment by mathomp4** [comment]
@rmontuoro I'm throwing a blocker label on this until I get the CI fixed. That'll need a pull from `main` due to #41. Note: this will also add the Changelog enforcer (like you've seen in MAPL). I've been lax in adding that.

File: ESMF/UFS/Aerosol_Comp_Mod.F90 [hunk 1]
```diff
@@ -40,6 +40,35 @@ module Aerosol_Comp_Mod
...
+  integer, parameter :: p_seas_3 = p_o3 + 16
+  integer, parameter :: p_seas_4 = p_o3 + 17
+  integer, parameter :: p_nh3    = p_o3 + 19
+  integer, parameter :: p_nh4a   = p_o3 + 20
+  integer, parameter :: p_no3an1 = p_o3 + 21
+  integer, parameter :: p_no3an2 = p_o3 + 22
+  integer, parameter :: p_no3an3 = p_o3 + 23
+  ! - diagnostic section (not advected)
+  integer, parameter :: p_pm25   = p_o3 + 24
+  integer, parameter :: p_pm10   = p_o3 + 25
```
Comment (tclune) [inline ESMF/UFS/Aerosol_Comp_Mod.F90:69]: perhaps consider using a Fortran enumerator instead?

Comment (tclune) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:69]: Ah - I now see this was just a move.   But probably still worth considering.

**Review by tclune — DISMISSED** [review]

**Review by gmao-esherman — DISMISSED** [review]

File: ESMF/UFS/Aerosol_Comp_Mod.F90 [hunk 2]
```diff
@@ -40,6 +40,35 @@ module Aerosol_Comp_Mod
...
+
+  integer, parameter :: p_seas_d = p_seas_1 - p_so2 - 3
```
Comment (adarmenov) [inline ESMF/UFS/Aerosol_Comp_Mod.F90:71]: I do not follow... what is the connection between seasalt and SO2?

File: ESMF/UFS/Aerosol_Comp_Mod.F90 [hunk 3]
```diff
@@ -709,6 +720,101 @@ subroutine AerosolTracerUpdate(state, cap, rc)
...
+      kk = nk - k + 1
+      do j = 1, nj
+        do i = 1, ni
+          ! -- compute partial PM2.5
+          pm25 = q(i,j,k,p_bc_1) + q(i,j,k,p_bc_2)     &
+               + q(i,j,k,p_oc_1) + q(i,j,k,p_oc_2)     &
+               + q(i,j,k,p_dust_1)                     &
+               + q(i,j,k,p_seas_1) + q(i,j,k,p_seas_2) &
+               + w_so4 * q(i,j,k,p_sulf)               &
+               + w_no3 * q(i,j,k,p_no3an1)
```
Comment (adarmenov) [inline ESMF/UFS/Aerosol_Comp_Mod.F90:795]: 1. @gmao-esherman Would this work if one or more aerosol types are disabled, dust for example?
2. @gmao-esherman I would expect that the PM diagnostics is calculated using only the active aerosols, is this the case?

File: ESMF/UFS/Aerosol_Comp_Mod.F90 [hunk 4]
```diff
@@ -709,6 +720,101 @@ subroutine AerosolTracerUpdate(state, cap, rc)
...
+
+    ! -- local parameters
+    real(ESMF_KIND_R8), parameter :: w_du2  = log(1.250_ESMF_KIND_R8) / log(1.8_ESMF_KIND_R8)
+    real(ESMF_KIND_R8), parameter :: w_du4  = log(1.667_ESMF_KIND_R8) / log(2.0_ESMF_KIND_R8)
+    real(ESMF_KIND_R8), parameter :: w_ss3  = log(2.50_ESMF_KIND_R8) / log(3._ESMF_KIND_R8)
+    real(ESMF_KIND_R8), parameter :: w_so4  = 132.14_ESMF_KIND_R8 / 96.06_ESMF_KIND_R8
+    real(ESMF_KIND_R8), parameter :: w_no3  = 80.043_ESMF_KIND_R8 / 62.0_ESMF_KIND_R8
+    real(ESMF_KIND_R8), parameter :: w25_no3an2 = 0.138_ESMF_KIND_R8 * w_no3
+    real(ESMF_KIND_R8), parameter :: w10_no3an2 = 0.808_ESMF_KIND_R8 * w_no3
+    real(ESMF_KIND_R8), parameter :: w10_no3an3 = 0.164_ESMF_KIND_R8 * w_no3
```
Comment (adarmenov) [inline ESMF/UFS/Aerosol_Comp_Mod.F90:746]: Is r8 really necessary?

File: ESMF/UFS/Aerosol_Comp_Mod.F90 (hunk 3 above)
Comment (gmao-esherman) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:795]: This is all under ESMF/UFS (NOAA stuff). Nothing under ESMF/UFS gets exercised by legacy GOCART or GOCART2G when we build GEOS.

Comment (adarmenov) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:795]: @gmao-esherman  Why are we asked to review it then, just a formality?

Comment (gmao-esherman) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:795]: Good question. I am not 100% sure why. I think any changes made to the @GOCART repository have to get reviewed by someone. Maybe @tclune has some more insight.

**Comment by tclune** [comment]
@adarmenov    I'm trying to fix it so you won't have to review.    Currently the CODEOWNERS file has you and Elliot reviewing changes anywhere under GOCART.   This one file should be managed separately.        Hopefully soon, but you just know it will happen after the last change needed by NOAA.

**Comment by mathomp4** [comment]
> @adarmenov I'm trying to fix it so you won't have to review. Currently the CODEOWNERS file has you and Elliot reviewing changes anywhere under GOCART. This one file should be managed separately. Hopefully soon, but you just know it will happen after the last change needed by NOAA.

Huh. How did @adarmenov get tagged on this. The CODEOWNER says:
```
/ESMF/UFS/              @tclune @weiyuan-jiang
```
Maybe I got this wrong...

File: ESMF/UFS/Aerosol_Comp_Mod.F90 (hunk 1 above)
Comment (rmontuoro) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:69]: @tclune - agreed. This should not be hardcoded and will hopefully be overhauled soon, perhaps using GFTL maps.
Are enumerators fully supported across Fortran compilers?

**Comment by tclune** [comment]
I don't use enumerators much, but the C-interoperability stuff is "easy" and I've not heard of any bug reports in that direction.    F202x will hopefully have a much more useful type of enumerator, as these are rather limited.

**Comment by tclune** [comment]
Certainly they are supported by all compilers that support current MAPL.

File: ESMF/UFS/Aerosol_Comp_Mod.F90 (hunk 2 above)
Comment (rmontuoro) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:71]: @adarmenov - `p_seas_d` is a computed offset to populate UFS-defined sea salt diagnostics. The only connection between sea salt and SO2 is their relative order in the tracer array.

File: ESMF/UFS/Aerosol_Comp_Mod.F90 (hunk 4 above)
Comment (rmontuoro) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:746]: @adarmenov - no. However, PM2.5 and PM10 are computed using tracer mixing ratios stored in an `ESMF_TYPEKIND_R8` coupling field, so `ESMF_KIND_R8` is being used for consistency.

File: ESMF/UFS/Aerosol_Comp_Mod.F90 (hunk 3 above)
Comment (rmontuoro) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:795]: @adarmenov, @gmao-esherman - Good point. As long as all required aerosol tracers are included in the atmospheric model, the PM computation will not be affected by one or more aerosol types being disabled. More flexibility might be added in the future.

File: ESMF/UFS/Aerosol_Comp_Mod.F90 (hunk 2 above)
Comment (adarmenov) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:71]: I can see that... This is very fragile in my opinion. Consider also the magical number 3, which I guess is the equivalent of (p_msa - p_so2).

Comment (rmontuoro) [reply ESMF/UFS/Aerosol_Comp_Mod.F90:71]: @adarmenov - all valid points. This code is slated to be refactored and diagnostics may be removed as implemented. The number 3 provides the offset for the exported diagnostic array, and it is not related to (p_msa-p_so2).

**Review by gmao-esherman — APPROVED** [review]

**Review by tclune — APPROVED** [review]

**Pull request merged**